app/controllers/restaurants_controller.rb

```diff
@@ -1,11 +1,15 @@
 class RestaurantsController < ApplicationController
-
   def index
     @restaurants = Restaurant.all
   end
-  
+
   def show
     @restaurant = Restaurant.find(params[:id])
   end
 
+  def rest_cat
```

Comment:
Yes, the trickiest part. We made an update to the code. See what you think.

Many thanks for your comments and remarks!